bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/MirroringConfiguration.java

```diff
@@ -0,0 +1,157 @@
+      this.primaryConfiguration = new Configuration(mirroringConfiguration.primaryConfiguration);
+      this.secondaryConfiguration =
+          new Configuration(mirroringConfiguration.secondaryConfiguration);
+      this.mirroringOptions = mirroringConfiguration.mirroringOptions;
+    } else {
+      checkParameters(conf);
+      this.primaryConfiguration = constructPrimaryConfiguration(conf);
+      this.secondaryConfiguration = constructSecondaryConfiguration(conf);
+      this.mirroringOptions = new MirroringOptions(conf);
+    }
+  }
+
+  private Configuration constructPrimaryConfiguration(Configuration conf) {
+    return constructConnectionConfiguration(
+        conf, MIRRORING_PRIMARY_CONNECTION_CLASS_KEY, MIRRORING_PRIMARY_CONFIG_PREFIX_KEY);
+  }
+
+  private Configuration constructSecondaryConfiguration(Configuration conf) {
+    return constructConnectionConfiguration(
+        conf, MIRRORING_SECONDARY_CONNECTION_CLASS_KEY, MIRRORING_SECONDARY_CONFIG_PREFIX_KEY);
+  }
+
+  private Configuration constructConnectionConfiguration(
+      Configuration conf, String connectionClassKey, String prefixKey) {
+    String connectionClassName = conf.get(connectionClassKey);
+    String prefix = conf.get(prefixKey, "");
+    Configuration connectionConfig = extractPrefixedConfig(prefix, conf);
+    connectionConfig.set("hbase.client.connection.impl", connectionClassName);
+    return connectionConfig;
+  }
+
+  private static void checkParameters(Configuration conf) {
+    String primaryConnectionClassName = conf.get(MIRRORING_PRIMARY_CONNECTION_CLASS_KEY);
+    String secondaryConnectionClassName = conf.get(MIRRORING_SECONDARY_CONNECTION_CLASS_KEY);
+    String primaryConnectionConfigPrefix = conf.get(MIRRORING_PRIMARY_CONFIG_PREFIX_KEY, "");
+    String secondaryConnectionConfigPrefix = conf.get(MIRRORING_SECONDARY_CONFIG_PREFIX_KEY, "");
+
+    checkArgument(
+        primaryConnectionClassName != null,
+        String.format("Specify %s.", MIRRORING_PRIMARY_CONNECTION_CLASS_KEY));
+    checkArgument(
+        secondaryConnectionClassName != null,
+        String.format("Specify %s.", MIRRORING_SECONDARY_CONNECTION_CLASS_KEY));
+
+    if (Objects.equals(primaryConnectionClassName, secondaryConnectionClassName)
+        && Objects.equals(primaryConnectionConfigPrefix, secondaryConnectionConfigPrefix)) {
+      if (primaryConnectionConfigPrefix.isEmpty()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Mirroring connections using the same client class requires a separate "
+                    + "configuration for one of them. Specify either %s or %s and use its value "
+                    + "as a prefix for configuration options.",
+                MIRRORING_PRIMARY_CONFIG_PREFIX_KEY, MIRRORING_SECONDARY_CONFIG_PREFIX_KEY));
+      } else {
+        throw new IllegalArgumentException(
+            String.format(
+                "Values of %s and %s should be different.",
```

Comment:
Can we use more descriptive message here and explain why we are demanding it to be different?

Reply:
OK, I've created an issue for this.